docs/FeatureManagement/Samples/TargetingFilter.tests.json

```diff
@@ -0,0 +1,211 @@
+    },
+    {
+        "TestID": "TargetingFilter-ComplexTargeting-TargetedUser",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Alice"},
+        "ExpectedResult": {
+            "IsEnabled": "true",
+            "Variant": {}
+        },
+        "Exception": {},
+        "Description": "Targeting Filter, Alice is a targeted user."
+    },
+    {
+        "TestID": "TargetingFilter-ComplexTargeting-TargetedGroup",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Aiden", "groups":["Stage1"]},
+        "ExpectedResult": {
+            "IsEnabled": "true",
+            "Variant": {}
+        },
+        "Exception": {},
+        "Description": "Targeting Filter, Aiden now part of the 100% rolled out group Stage1 is now targeted."
+    },
+    {
+        "TestID": "TargetingFilter-ComplexTargeting-DisabledTargetedGroup",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"groups":["Stage2"]},
+        "ExpectedResult": {
+            "IsEnabled": "false",
+            "Variant": {}
+        },
+        "Exception": {},
+        "Description": "Targeting Filter, no user Stage 2 is 50% rolloued out."
+    },
+    {
+        "TestID": "TargetingFilter-ComplexTargeting-EnabledTargetedGroup50",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Aiden", "groups":["Stage2"]},
+        "ExpectedResult": {
+            "IsEnabled": "true",
+            "Variant": {}
+        },
+        "Exception": {},
+        "Description": "Targeting Filter, Aiden who is not part of the default rollout is part of the first 50% of Stage 2."
+    },
+    {
+        "TestID": "TargetingFilter-ComplexTargeting-DisabledTargetedGroup50",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Chad", "groups":["Stage2"]},
+        "ExpectedResult": {
+            "IsEnabled": "false",
+            "Variant": {}
+        },
+        "Exception": {},
+        "Description": "Targeting Filter, Chad is neither part of the default rollout or part of the first 50% of Stage 2."
+    },
+    {
+        "TestID": "TargetingFilter-ComplexTargeting-ExcludedGroup",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"groups":["Stage3"]},
+        "ExpectedResult": {
+            "IsEnabled": "false",
+            "Variant": {}
+        },
+        "Exception": {},
+        "Description": "Targeting Filter, the Stage 3 is group is on the exclusion list."
+    },
+    {
+        "TestID": "TargetingFilter-ComplexTargeting-ExcludedGroupTargetedUser",
+        "FeatureFlagName": "ComplexTargeting",
+        "Inputs": {"user":"Alice", "groups":["Stage3"]},
+        "ExpectedResult": {
+            "IsEnabled": "false",
+            "Variant": {}
+        },
+        "Exception": {},
+        "Description": "Targeting Filter, Alice who is targeted is excluded because she is part of the Stage 3 group."
```

Comment:
How about this: Alice is excluded because she is part of the Stage 3 group, even if she is an included user. Exclusion rules have higher priority than inclusion rules.

BTW, I feel like we should introduce the targeting evaluation flow somewhere . It can be a chart or comments. 
This will be difficult for people have no context about targeting evaluation to follow.

Reply:
These files are mainly validation files for our libraries. We shouldn't be explaining usage here, at most we redirect them to the actually libraries that already explain the logic.

Reply:
I am all for making these samples and expected results as clear as possible for us and customers, but they are not replacement for documentation. I don't expect someone to learn feature management solely from these samples.